### ISSUE TITLE
Cannot locate package MultiArrayLayout

### ISSUE THREAD

**Issue by amittleider** [ros/ros_comm#238]
ROS Bag cannot utilize the MultiArrayLayout message.  It gives the following stack trace:

Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 551, in __bootstrap_inner
    self.run()
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rxbag/timeline.py", line 1923, in run
    msg_data = self._get_message(bag, playhead_position)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rxbag/timeline.py", line 1936, in _get_message
    msg_data = self.timeline.read_message(bag, position)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rxbag/timeline.py", line 708, in read_message
    return bag._read_message(position)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rosbag/bag.py", line 774, in _read_message
    return self._reader.seek_and_read_message_data_record(position, raw)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rosbag/bag.py", line 2198, in seek_and_read_message_data_record
    msg_type = _get_message_type(connection_info)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/rosbag/bag.py", line 1309, in _get_message_type
    message_type = genpy.dynamic.generate_dynamic(info.datatype, info.msg_def)[info.datatype]
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genpy/dynamic.py", line 151, in generate_dynamic
    for l in msg_generator(msg_context, spec, search_path):
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genpy/generator.py", line 725, in msg_generator
    genmsg.msg_loader.load_depends(msg_context, spec, search_path)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genmsg/msg_loader.py", line 344, in load_depends
    return load_msg_depends(msg_context, spec, msg_search_path)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genmsg/msg_loader.py", line 313, in load_msg_depends
    depspec = load_msg_by_type(msg_context, resolved_type, search_path)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genmsg/msg_loader.py", line 119, in load_msg_by_type
    file_path = get_msg_file(package_name, base_type, search_path)
  File "/opt/ros/fuerte/lib/python2.7/dist-packages/genmsg/msg_loader.py", line 78, in get_msg_file
    % (base_type, package, search_path))
MsgNotFound: Cannot locate message [MultiArrayLayout]: unknown package [std_msgs] on search path [{}]

The error says that it is a problem with my path, but it most definitely is not a problem with the path.  First of all, in the exact same bag, there are many messages in the std_msgs package (which work fine).  Secondly, you can notice that the search path is the empty set, which is clearly incorrect since it should, by default, include all the default ROS packages that come with a clean installation.

The problem is easy to reproduce.  Simply publish any message of type Int16MultiArray, Int32MultiArray, or their Float counterparts and record it to a bag file.  Next, open the bag file with the rxbag tool.  Then right click -> View by topic -> Your MultiArray topic -> Raw or Plot.  You will notice that the tool does not pull up data and your terminal gives the stack trace that i have posted above.

Note that this problem also exists when trying to parse a ROS bag with the python/c++ bag API.

**Comment by dirk-thomas** [comment]
I just created a publisher for Int32MultiArray messages, recorded the topic with rosbag, and was able to play it back with rosbag and rxbag. The raw view is empty until you hit play for the bag and the first message is played back than it shows the message data. There are no error messages / stack traces on the console. I tried it on Fuerte and Groovy.

Can you provide more information on that? The exact versions of packages you are using. A step-by-step command sequence which reproduces the issue for you.

**Comment by amittleider** [comment]
Thanks for the reply, Dirk.

The ROS version is Fuerte 1.8.10, std_msgs 0.4.11, ros_comm 1.8.15.

A step by step command sequence would be anything that reads a bag.  That's
why I suggested checking it on rxbag.  I did know about the little quirk in
rxbag, but this is not the same issue.

I have also written a bag parser using the rosbag python API, and I receive
the same stack trace error message.  I was able to reproduce the problem on
a co-worker's computer.  We are likely running the same version of ROS.

As a quick fix, maybe I could simply upgrade.  Although I'd rather not
upgrade to Groovy yet if I can avoid it.  Maybe simply upgrading ros_comm
can fix it??

Thanks
-Andrew

**Comment by dirk-thomas** [comment]
The new versions for Fuerte of ros/ros_comm will pretty sure make no difference - they contain no changes related to rosbag. You can give it a try just to guarantee that we both have the same setup.

You can not cross-distro upgrade any packages.

I think the next step should be that I run the exact command sequence on my machine and check if I still don't get the error. Without being able to reproduce it becomes nearly impossible to suggest any changes. Can you provide that to make sure that I am not doing slightly different steps?

**Comment by amittleider** [comment]
I double checked your steps to open a bag in rxbag, they are the same as
mine, but you are not receiving any errors.  I repeated the steps that you
did, publishing a Int16MultiArray message and making a rosbag, then opening
it with rxbag.  I was able to view the messages in rxbag with ROS CPP and
ROS Python, but not ROS Java.  That means this problem is specific to ROS
Java.  Sorry for the confusion.
